tests/unit/test_mac_frame.cpp

```diff
@@ -787,6 +787,118 @@ void TestMacFrameAckGeneration(void)
 #endif // (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
 }
 
+void TestMacWakeupFrameGeneration(void)
+{
+#if OPENTHREAD_CONFIG_WAKEUP_COORDINATOR_ENABLE
+    constexpr uint16_t kMpFcfLongFrame           = 1 << 3;
+    constexpr uint16_t kMpFcfDstAddrShift        = 4;
+    constexpr uint16_t kMpFcfDstAddrExt          = 3 << kMpFcfDstAddrShift;
+    constexpr uint16_t kMpFcfSrcAddrShift        = 6;
+    constexpr uint16_t kMpFcfSrcAddrExt          = 3 << kMpFcfSrcAddrShift;
+    constexpr uint16_t kMpFcfPanidPresent        = 1 << 8;
+    constexpr uint16_t kMpFcfSecurityEnabled     = 1 << 9;
+    constexpr uint16_t kMpFcfSequenceSuppression = 1 << 10;
+    constexpr uint16_t kMpFcfIePresent           = 1 << 15;
+
+    constexpr static uint8_t srcExtaddr[] = {0x55, 0x55, 0x55, 0x55, 0x55, 0x55, 0x55, 0x55};
+    constexpr static uint8_t dstExtaddr[] = {0xDD, 0xDD, 0xDD, 0xDD, 0xDD, 0xDD, 0xDD, 0xDD};
+    constexpr static uint8_t keySource[]  = {0, 0, 0, 0x1C};
+
+    // clang-format off
+    constexpr static uint8_t wakeupPsdu[] = {
+        // Frame Control
+        Mac::Frame::kTypeMultipurpose | kMpFcfLongFrame | kMpFcfDstAddrExt | kMpFcfSrcAddrExt,
+        (kMpFcfPanidPresent | kMpFcfSecurityEnabled | kMpFcfSequenceSuppression | kMpFcfIePresent) >> 8,
+        // PAN ID
+        0xCE,
+        0xFA,
+        // Destination Address
+        0xDD,
+        0xDD,
+        0xDD,
+        0xDD,
+        0xDD,
+        0xDD,
+        0xDD,
+        0xDD,
+        // Source Address
+        0x55,
+        0x55,
+        0x55,
+        0x55,
+        0x55,
+        0x55,
+        0x55,
+        0x55,
+        // Security Header
+        Mac::Frame::kKeyIdMode2 | Mac::Frame::kSecurityEncMic32,
+        0xFC,
+        0xFC,
+        0xFC,
+        0xFC,
+        0x00,
+        0x00,
+        0x00,
+        0x1C,
+        0x1D,
+        // Rendezvous Time IE
+        0x82,
+        0x0E,
+        0xCD,
+        0xAB,
+        // Connection IE
+        0x05,
+        0x00,
+        0x9B,
+        0xB8,
+        0xEA,
+        0x01,
+        0x1C, //< Connection Window
```

Comment:
I guess `clang-format` is forcing this format. May look nice if we have change it to 
```c++
   Mac::Frame::kKeyIdMode2 | Mac::Frame::kSecurityEncMic32, 0xFC, 0xFC, 0xFC, 0xFC, 0x00, 
   0x00, 0x00, 0x1C, 0x1D,
```


I see you disabled `clang-format`? does it still try to arrange the numbers?